dexcom/default.go

```diff
@@ -4,14 +4,20 @@ import (
 	"fmt"
 
 	"github.com/tidepool-org/platform/pointer"
+	"github.com/tidepool-org/platform/structure"
 )
 
 // NOTE: currently we are finding that g7 data occasionally doesn't have the required units
 // data. This is a workaround until that issue is resolved
-func StringOrDefault(givenValue *string, defaultValue interface{}) *string {
+func StringOrDefault(parser structure.ObjectParser, reference string, defaultValue interface{}) *string {
+	givenValue := parser.String(reference)
 	if givenValue == nil && defaultValue != nil {
```

Comment:
I think we may need to expand this to also detect an empty string value (i.e. "") - in addition to the `nil`. Unless the string parser turns an empty string value into a nil?